### PR TITLE
add test case: test set policy with condition operator end with IfExists

### PR DESCRIPTION
**Pull request by joke-lee**
add test case for https://github.com/ceph/ceph/pull/15607

### CODE REVIEW

File: s3tests/functional/test_s3.py [hunk 1]
```diff
@@ -8476,3 +8476,28 @@ def test_sse_kms_barb_transfer_13b():
...
+        "Condition": {
+                    "StringLikeIfExists": {
+                        "aws:Referer": "http://www.example.com"
+                    }
+                },
+        "Resource": "arn:aws:s3:::%s/*"
+      }
+     ]
+    }''' % bucket.name
+    eq(bucket.set_policy(policy), True)
```
Comment (adamemerson) [inline s3tests/functional/test_s3.py]: This looks good. Though it might be good to add a request in with a fake referer header, too.

Comment (joke-lee) [inline s3tests/functional/test_s3.py]: hi @adamemerson , i have add  some requests with fake referer. can you help to review it ? thanks

Comment (adamemerson) [inline s3tests/functional/test_s3.py]: I am pleased with this

**Comment by joke-lee** [comment]
@cbodley  would you mind to help review this please? thanks

**Comment by cbodley** [comment]
@joke-lee looks good. i don't have write access, we'll need @yehudasa to merge

**Comment by joke-lee** [comment]
@yehudasa  do you mind help to review this ?

**Comment by joke-lee** [comment]
hi @mattbenjamin do you mind help to review this ?

**Comment by joke-lee** [comment]
hi @yehudasa ,do you mind looking this ?

**Comment by yehudasa** [comment]
@joke-lee can you rebase? thanks

**Comment by joke-lee** [comment]
hi @yehudasa , rebased done

**Pull request merged**